### PR TITLE
Fix update conf error for int,double,boolean

### LINKED ISSUE #15816: Update configuration command line tools break to update boolean
**Alluxio Version:**
Master branch & 2.8.2

**Describe the bug**

```
java.lang.IllegalArgumentException: Invalid value for property key XXX: true
	at com.google.common.base.Preconditions.checkArgument(Preconditions.java:435)
	at alluxio.conf.InstancedConfiguration.set(InstancedConfiguration.java:246)
	at alluxio.conf.ServerConfiguration.set(ServerConfiguration.java:112)
	at alluxio.master.meta.DefaultMetaMaster.updateConfiguration(DefaultMetaMaster.java:618)
	at alluxio.master.meta.MetaMasterConfigurationServiceHandler.lambda$updateConfiguration$5(MetaMasterConfigurationServiceHandler.java:139)
	at alluxio.RpcUtils.callAndReturn(RpcUtils.java:122)
	at alluxio.RpcUtils.call(RpcUtils.java:83)
	at alluxio.RpcUtils.call(RpcUtils.java:58)
	at alluxio.master.meta.MetaMasterConfigurationServiceHandler.updateConfiguration(MetaMasterConfigurationServiceHandler.java:137)
	at alluxio.grpc.MetaMasterConfigurationServiceGrpc$MethodHandlers.invoke(MetaMasterConfigurationServiceGrpc.java:585)
	at io.grpc.stub.ServerCalls$UnaryServerCallHandler$UnaryServerCallListener.onHalfClose(ServerCalls.java:182)
	at io.grpc.internal.ServerCallImpl$ServerStreamListenerImpl.halfClosed(ServerCallImpl.java:331)
	at io.grpc.internal.ServerImpl$JumpToApplicationThreadServerStreamListener$1HalfClosed.runInContext(ServerImpl.java:797)
	at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37)
	at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:123)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
```

**To Reproduce**

```bash
bin/alluxio fsadmin updateConf xxx=true
```

**Expected behavior**

Update conf successfully

**Urgency**

Urgent.

**Are you planning to fix it**

Yeah

**Additional context**
No

### PR DESCRIPTION
**Pull request by maobaolong**
Fix #15816

### CODE REVIEW

File: core/server/master/src/main/java/alluxio/master/meta/DefaultMetaMaster.java [hunk 1]
```diff
@@ -647,6 +648,25 @@ public Map<String, Boolean> updateConfiguration(Map<String, String> propertiesMa
     return result;
   }
 
+  private Object getTypedValue(Map.Entry<String, String> entry, PropertyKey key) {
```
Comment (apc999) [inline core/server/master/src/main/java/alluxio/master/meta/DefaultMetaMaster.java:651]: shall we put this into Configuration related classes? It looks useful to me

File: core/server/master/src/main/java/alluxio/master/meta/DefaultMetaMaster.java [hunk 2]
```diff
@@ -629,7 +629,8 @@ public Map<String, Boolean> updateConfiguration(Map<String, String> propertiesMa
         if (Configuration.getBoolean(PropertyKey.CONF_DYNAMIC_UPDATE_ENABLED)
             && key.isDynamic()) {
           Object oldValue = Configuration.get(key);
-          Configuration.set(key, entry.getValue(), Source.RUNTIME);
+          Object value = getTypedValue(entry, key);
```
Comment (rongrong) [inline core/server/master/src/main/java/alluxio/master/meta/DefaultMetaMaster.java:632]: why not just use `key.parseValue` here?

File: core/common/src/main/java/alluxio/conf/InstancedConfiguration.java [hunk 3]
```diff
@@ -215,9 +215,9 @@ public void set(@Nonnull PropertyKey key, @Nonnull Object value, @Nonnull Source
     checkArgument(!value.equals(""),
         "The key \"%s\" cannot be have an empty string as a value. Use "
             + "Configuration.unset to remove a key from the configuration.", key);
+    value = key.formatValue(value);
```
Comment (rongrong) [inline core/common/src/main/java/alluxio/conf/InstancedConfiguration.java:218]: why is this change needed?

File: core/common/src/main/java/alluxio/conf/PropertyKey.java [hunk 4]
```diff
@@ -8681,6 +8681,15 @@ private static Object formatValue(Object value, PropertyType type,
           case DATASIZE:
             FormatUtils.parseSpaceSize(stringValue);
             break;
+          case BOOLEAN:
```
Comment (rongrong) [inline core/common/src/main/java/alluxio/conf/PropertyKey.java:8684]: Why is this change needed?

File: core/server/master/src/main/java/alluxio/master/meta/DefaultMetaMaster.java (hunk 2 above)
Comment (maobaolong) [reply core/server/master/src/main/java/alluxio/master/meta/DefaultMetaMaster.java:632]: Thanks.

File: core/server/master/src/main/java/alluxio/master/meta/DefaultMetaMaster.java (hunk 1 above)
Comment (maobaolong) [reply core/server/master/src/main/java/alluxio/master/meta/DefaultMetaMaster.java:651]: Yeah

File: core/common/src/main/java/alluxio/conf/InstancedConfiguration.java (hunk 3 above)
Comment (maobaolong) [reply core/common/src/main/java/alluxio/conf/InstancedConfiguration.java]: Thanks

File: core/common/src/main/java/alluxio/conf/PropertyKey.java (hunk 4 above)
Comment (maobaolong) [reply core/common/src/main/java/alluxio/conf/PropertyKey.java]: Thanks

**Comment by maobaolong** [comment]
@rongrong PTAL.

**Comment by maobaolong** [comment]
@apc999 @rongrong Hi, would you please take a look at this PR, it seems more simple and tiny than before, thanks for the suggestion from you!

**Review by apc999 — APPROVED** [review]

**Comment by maobaolong** [comment]
alluxio-bot, merge this please

**Pull request merged**